platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/address/AddressBookUtils.java

```diff
@@ -239,9 +239,9 @@ public static Address parseAddressText(@NonNull final String addressText) throws
                 nickname,
                 selfname,
                 weight,
-                internalIp.getHostAddress(),
+                parts[5], // FQDN Support: use original string for internal address
```

Comment:
Is there a reason that you stopped using `internalIp` and `externalIp`, but didn't remove their parsing from the method?

Reply:
The parsing generates a network `UnknownHostException` if the text is not network resolvable.   I wanted to leave that unchanged.    Does the parsing really do anything?  I don't know.  It made sense at the time to make sure that the hostname was network resolvable.  Perhaps it doesn't make sense here and failure at the network connection attempt is more appropriate.    What do you think?

Reply:
I think you're right, it makes sense to make sure it resolves. 

I think something like the following would be good, so that the intention is clear. Otherwise, someone is likely to remove the block in the future, thinking it's no longer necessary

```
        try {
            // make sure the IP address is valid, even though we don't use the result
            final InetAddress ignored = InetAddress.getByName(parts[5]);
        } catch (UnknownHostException e) {
            throw new ParseException("Cannot parse ip address from '" + parts[5] + ",", 5);
        }
```